### PR TITLE
chore(cli): add `default_client_version` to rethCli

### LINKED ISSUE #11768: Add client_version to RethCli
## TODO

* add a `fn client_version() -> ClientVersion` to 

https://github.com/paradigmxyz/reth/blob/7b1b1fcb3b632d8c877a3e761a77ae4d513c0c2a/crates/cli/cli/src/lib.rs#L24-L24

ref:

https://github.com/paradigmxyz/reth/blob/7b1b1fcb3b632d8c877a3e761a77ae4d513c0c2a/crates/node/core/src/version.rs#L132-L132

### PR DESCRIPTION
**Pull request by aroralanuk**
added a `default_client_version` in rethCli which references the same version used in node_core crate.

Fixes: https://github.com/paradigmxyz/reth/issues/11768

### CODE REVIEW

File: crates/cli/cli/src/lib.rs [hunk 1]
```diff
@@ -66,4 +68,9 @@ pub trait RethCli: Sized {
 
         Ok(cli.with_runner(f))
     }
+
+    /// The default client version accessing the database. ref: reth_node_core::version::default_client_version
+    fn default_client_version() -> ClientVersion {
+        default_client_version()
+    }
```
Comment (mattsse) [inline crates/cli/cli/src/lib.rs:75]: this should just be
```suggestion
    /// The client version of the node.
    fn client_version() -> ClientVersion;
```

Comment (mattsse) [reply crates/cli/cli/src/lib.rs:75]: so that the implementer can configure this

**Review by mattsse — CHANGES_REQUESTED** [review]

**Review by mattsse — APPROVED** [review]

**Comment by aroralanuk** [comment]
Sorry, I should've ran clippy before asking for review :|

**Comment by mattsse** [comment]
no worries after this it should also be run automatically

**Pull request merged**